runtime/src/main/java/org/corfudb/runtime/collections/CorfuStore.java

```diff
@@ -433,11 +436,13 @@ public void subscribeLogReplicationListener(@Nonnull LogReplicationListener stre
      * @param namespace        namespace of the replicated tables
      * @param streamTag        stream tag of the replicated tables
      * @param bufferSize       maximum size of buffered transaction entries
+     * @param executorService  thread pool to perform client full sync asynchronously from subscription
      */
     public void subscribeLogReplicationListener(@Nonnull LogReplicationListener streamListener,
-                                                @Nonnull String namespace, @Nonnull String streamTag, int bufferSize) {
+                                                @Nonnull String namespace, @Nonnull String streamTag,
+                                                int bufferSize, ExecutorService executorService) {
```

Comment:
Just a thought can we accept a `ScheduledExecutorService` instead of `ExecutorService` this allows us to have more flexibility in setting up scheduled tasks too later?